<a href="https://colab.research.google.com/github/mohandaz/HIA-302-Project/blob/main/HIA302_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

CLone Github Repository : https://github.com/mohandaz/HIA-302-Project

In [1]:
!git clone https://github.com/mohandaz/HIA-302-Project

Cloning into 'HIA-302-Project'...
remote: Enumerating objects: 50, done.
remote: Counting objects: 100% (50/50), done.
remote: Compressing objects: 100% (46/46), done.
remote: Total 50 (delta 22), reused 0 (delta 0), pack-reused 0
Receiving objects: 100% (50/50), 280.90 KiB | 9.06 MiB/s, done.
Resolving deltas: 100% (22/22), done.


Libraries

In [2]:
import pandas as pd
import csv
import numpy as np

2.1 Demonstrate different ways to convert the horse-colic.txt file to horse-colic.csv.

In [13]:
import csv
import pandas as pd

# Specify input and output directories
input_dir = '/content/HIA-302-Project/horse-colic.txt'
output_dir = '/content/HIA-302-Project/horse-colic.csv'

# Read the text file into a DataFrame
df = pd.read_csv(input_dir, sep='\s+', header=None)

# Write to CSV file with empty header and default index
with open(output_dir, 'w', newline='') as csv_file:
    csv_writer = csv.writer(csv_file)

    # Write an empty row as the header
    csv_writer.writerow([])

    # Write the data rows
    for _, row in df.iterrows():
        csv_writer.writerow(row)



Converting .txt to .csv - using Python built-in 'csv' module

In [11]:
import csv

# Input and output file paths
txt_path = "/content/HIA-302-Project/horse-colic.txt"
csv_path = "/content/HIA-302-Project/horse-colic.csv"

# Read lines from the text file
with open(txt_path, 'r') as txt_file:
    lines = txt_file.readlines()

# Extract data lines
data_lines = [line.strip().split(',') for line in lines]

# Write to CSV file with empty header and default index
with open(csv_path, 'w', newline='') as csv_file:
    csv_writer = csv.writer(csv_file)

    # Write data
    csv_writer.writerows(data_lines)

print(f"Conversion successful. CSV file saved at: {csv_path}")

Conversion successful. CSV file saved at: /content/HIA-302-Project/horse-colic.csv


Checking data types

Question 2.2
 : Demonstrate that the csv file can read in Jupyter Notebook, and the dataset remains the same as the raw dataset.


In [ ]:
# Specify the path to the CSV file
csv_dir = '/content/HIA-302-Project/horse-colic.csv'

# Read the CSV file into a Pandas DataFrame
horse_colic = pd.read_csv(csv_dir, index_col=0, header=0)

# Display the first few rows of the CSV DataFrame

horse_colic.info()
horse_colic.head()


2.3 Replase the missing values (i.e. with"?" with NaN (Not a Number) value in a loaded dataset using python

In [272]:
horse_colic_with_NaN = horse_colic.replace("?","NaN")



In [ ]:
horse_colic_with_NaN.head()

2.4 Save the latest dataset with NaN as horse_colic_with_Nan.csv and ensure the data frame's row number is not saved into csv file

In [275]:
output_dir = '/content/HIA-302-Project/horse_colic_with_NaN.csv'
# Save the DataFrame to a CSV file
horse_colic_with_NaN.to_csv(output_dir)
horse_colic_with_NaN.to_csv(output_dir, index=False)

print(f"CSV file saved at: {output_dir}")

CSV file saved at: /content/HIA-302-Project/horse_colic_with_NaN.csv


In [257]:

print(f"Total number of NaN values in the DataFrame: {total_nan_count}")

Replacement completed. CSV file saved.
Number of NaN values in each column:
0       1
1       0
2       0
3      60
4      24
5      58
6      56
7      69
8      47
9      32
10     55
11     44
12     56
13    104
14    106
15    247
16    102
17    118
18     29
19     33
20    165
21    198
22      1
23      0
24      0
25      0
26      0
27      0
dtype: int64
Total number of NaN values in the DataFrame: 1605


2.5 Rename the header by following the description file (horse-colic.names) with not more than 10 characters i the header title

2.6 Save the latest dataset with header title as horse_colic_with_title.csv and ensure the data frame's row number is not saved into csv file.

2.7 Load the horse_colic_with_title.csv file and display list of total NaN at each column in the data set.

2.8 Visualize the totsl NaN at each column in the dataset with a bar chart and set colour to 'blue' using Metplotlib library

2.9 Retrive the datatypes for the dataset

2.10 In this dataset, will it be good idea to delete/drop tha NaN rows? Justify your answer

2.11 Read and evaluate the following codes in Figure 1. What method is the code using to reduce missing values from 1605 to 0? Justify your answer

2.12 Fill out the inline comments marked with hashtags # in lines 8,11,17,20,22. and 26. include the missing codes in line 5 and 6